plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/DataflowDependencyManager.java

```diff
@@ -0,0 +1,223 @@
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.eclipse.dataflow.core.project;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.versioning.ArtifactVersion;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.apache.maven.artifact.versioning.InvalidVersionSpecificationException;
+import org.apache.maven.artifact.versioning.VersionRange;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.m2e.core.MavenPlugin;
+import org.eclipse.m2e.core.embedder.IMaven;
+import org.eclipse.m2e.core.project.IMavenProjectFacade;
+import org.eclipse.m2e.core.project.IMavenProjectRegistry;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.NavigableSet;
+import java.util.TreeMap;
+
+/**
+ * {@link DataflowDependencyManager} provides {@code Dependency} instances for Dataflow and the
+ * Dataflow Examples.
+ */
+public class DataflowDependencyManager {
+  private final DataflowArtifactRetriever artifactRetriever;
+
+  private final IMaven maven;
+  private final IMavenProjectRegistry mavenProjectRegistry;
+
+  public static DataflowDependencyManager create() {
+    return create(
+        DataflowArtifactRetriever.defaultInstance(),
+        MavenPlugin.getMaven(),
+        MavenPlugin.getMavenProjectRegistry());
+  }
+
+  @VisibleForTesting
+  static DataflowDependencyManager create(
+      DataflowArtifactRetriever artifactRetriever,
+      IMaven maven,
+      IMavenProjectRegistry mavenProjectRegistry) {
+    return new DataflowDependencyManager(artifactRetriever, maven, mavenProjectRegistry);
+  }
+
+  private DataflowDependencyManager(
+      DataflowArtifactRetriever artifactRetriever,
+      IMaven maven,
+      IMavenProjectRegistry mavenProjectRegistry) {
+    this.artifactRetriever = artifactRetriever;
+    this.maven = maven;
+    this.mavenProjectRegistry = mavenProjectRegistry;
+  }
+
+  /**
+   * Retrieves a dependency on the Dataflow Java SDK. If trackUpdates is true, the version is
+   * LATEST. Otherwise, the version is [Current Version, Next Major Version).
```

Comment:
This is good to know. Mostly, this can be fixed by just removing the `trackDataflowDependency` method in `DataflowMavenModel` and its users.